pyrometheus/__init__.py

```diff
@@ -352,10 +350,15 @@ def _pyro_make_array(res_list):
     \"""
 
     from numbers import Number
-    all_numbers = all(isinstance(e, Number) for e in res_list)
+    all_numbers = all(
+        isinstance(e, Number)
+        or (isinstance(e, usr_np.ndarray) and e.shape == ())
+        for e in res_list)
```

Comment:
We want to be careful here. The interface documentation for the user's numpy clone does not indicate that it should sport the (.ndarray), (.dot), or (.zeros_like) members.  I'm not sure that the [numpy clone object](https://github.com/inducer/meshmode/blob/ac37ba5bc53223f40c195e25b23778cf6a004329/meshmode/array_context.py#L59) we actually use has these features.

Reply:
FWIW, this the extra `isinstance` check is needed because `np.exp(10)` returns a scalar, but `jnp.exp(10)` returns jax's `ndarray` lookalike. Not quite sure how to unify that.

Reply:
> Maybe doing `getattr(e, 'shape', None) == ()` is enough?

Not loving our options here. I *might* be OK with this. As an alternate option, could we wrap `jnp` before passing it in as `usr_np` to incur the cost of this only with it?

Reply:
Yeah, I also think wraping `jnp` a bit is the clearer way to go here. There are probably other differences that will pop up.

There's a `jax.device_get` that should get a nice number ouf of jax's `DeviceArray`, if necessary (pretty much the same as `pyopencl.array.Array.get`).

Reply:
So apparently, indexing into a `jax.numpy` array full of scalars will give back an array of shape `()` rather than the number itself. I have been trying to create a wrapper for `jnp` but bearing that issue in mind, I'm not sure if this is something wrapping can fix...

Reply:
This is addressed differently by 8ccf61b, moving these functions that check for numbers into the generated class so they can be overwritten when Jax is in use, in a manner similar to https://github.com/ecisneros8/pyrometheus/pull/25.